src/array.jl

```diff
@@ -54,3 +54,22 @@ function _prox!{T <: Number}(p::GroupLassoPenalty{T}, A::AA{T, 1}, s::T)
     end
     A
 end
+
+#-----------------------------------------------------------------# MahalanobisPenalty
+"""
+    MahalanobisPenalty(C)
+
+Supports a Mahalanobis distance penalty (`xᵀCᵀCx` for a vector `x`).
+"""
+type MahalanobisPenalty{T <: Number} <: ArrayPenalty
+    λ::T
+    C::AA{T,2}
+end
+MahalanobisPenalty{T}(C::AA{T}) = MahalanobisPenalty(one(T),C)
+
+value{T <: Number}(p::MahalanobisPenalty{T}, x) = T(0.5)*p.λ*sumabs2(p.C*x)
+
+function _prox!{T <: Number}(p::MahalanobisPenalty{T}, A::AA{T, 1}, s::T)
+    y = (p.C'p.C + (one(T)/s)*I) \ (A./s)
```

Comment:
Is it worth storing `C'C` in the type (or a buffer for `C'C + I/s`)?  I've never used this kind of penalty before.  What are "typical" dimensions of `C`?

It doesn't need to be changed now, but `C'C` and solving the system could be faster with `BLAS.syrk!` and `LAPACK.sysv!` (if we have a buffer to work with).